src/download.rs

```diff
@@ -19,6 +21,17 @@ where
         .send()
         .await?;
 
+    // Return immediately on download failure on the client side.
+    let status = res.status();
+    if status.is_client_error() {
```

Comment:
```suggestion
    if status.is_client_error() || status.is_server_error() {
```

Reply:
We should also either make use of https://docs.rs/reqwest/latest/reqwest/redirect/struct.Attempt.html#method.follow or check `is_redirection` and act accordingly.

Reply:
Updated to `!status.is_success()`.

Created a new issue for the redirect issue, https://github.com/flatcar/ue-rs/issues/27.
That looks out of scope of this PR. Probably needs more work, which I would like to add after we merge pending PRs including anyhow.

Reply:
Thanks!